langkit/support/langkit_support-vectors.adb

```diff
@@ -50,22 +50,34 @@ package body Langkit_Support.Vectors is
    procedure Reserve (Self : in out Vector; Capacity : Positive) is
       Siz : constant size_t := size_t (Capacity) * El_Size;
    begin
-      if Small_Vector_Capacity > 0 then
-         if Self.Capacity = Small_Vector_Capacity then
-            Self.E := To_Pointer (Alloc (Siz));
-            for I in Self.SV'Range loop
-               Self.E.all (I) := Self.SV (I);
-            end loop;
-         else
-            Self.E := To_Pointer (Realloc (Self.E.all'Address, Siz));
+      if Small_Vector_Capacity > 0
+        and then Self.Capacity = Small_Vector_Capacity
```

Comment:
Just in case: have you checked the performance impact of this? IIRC, we had two separate IF blocks because when the vector is instanciated with `Small_Vector_Capacity => 0`, GCC would fold the always false `Small_Vector_Capacity > 0` condition only if it was a standalone condition in an IF block.

Reply:
I didn't but I'm really dubious that this is why the code was written that way, I think it was written that way for correctness reasons in that particular case (the `Self.Capacity := Capacity` being at the end)

Reply:
Just checked: no performance difference